internal/rate/sliding_log.go

```diff
@@ -0,0 +1,144 @@
+package rate
+
+import (
+	"context"
+	"strconv"
+	"time"
+
+	"github.com/go-redis/redis/v8"
+
+	"github.com/TykTechnologies/tyk/storage"
+)
+
+type SlidingLog struct {
+	conn     redis.UniversalClient
+	pipeline bool
+}
+
+func NewSlidingLog(cluster storage.Handler, pipeline bool) (*SlidingLog, error) {
+	conn := new(redis.UniversalClient)
+	if err := cluster.As(conn); err != nil {
+		return nil, err
+	}
+
+	return NewSlidingLogRedis(*conn, pipeline), nil
+}
+
+func NewSlidingLogRedis(conn redis.UniversalClient, pipeline bool) *SlidingLog {
```

Comment:
should wemake this function only pkg scoped? and expose only NewSlidingLog?

Reply:
@sredxny I'm more inclined to remove the default one. I already implemented it as an interface so i can fully drop the storage package import (a lot of that package needs to be kicked out).